.github/workflows/check-external-links.yaml

```diff
@@ -0,0 +1,40 @@
+name: Check external links
+
+on: workflow_dispatch
+
+jobs:
+  check-links:
+    name: Check for broken external links
+    runs-on: 'ubuntu-latest'
+
+    strategy:
+      fail-fast: false
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4.2.1
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4.0.4
+        with:
+          cache: npm
+          check-latest: true
+          node-version-file: .nvmrc
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Restore build
+        uses: actions/cache/restore@v4.1.1
+        with:
+          key: build-cache-${{ runner.os }}-${{ github.sha }}
+          path: build
```

Comment:
**issue** This makes the workflow dependent on the build existing in GitHub actions' cache. [Caches expire after 7 days without being used](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy). 

I think we need to look at sharing the `build` job between the `test` and `check-external-links` workflow. This is [something we do in `govuk-frontend`](https://github.com/alphagov/govuk-frontend/blob/main/.github/workflows/tests.yml#L61-L62) if we need 'inspiration' 😉

Reply:
@romaricpascal I've had a go at this in https://github.com/alphagov/govuk-design-system/pull/4235/commits/88178a8d15d75040bef9d0ec856582b8542d0007. I'm not feeling ultra confident about it, especially since js tests are failing and specifically calling out bad cahcing, which I suspect is the problem. Any advice?